zam/aspirateur/zam_aspirateur/senateurs/parse.py

```diff
@@ -0,0 +1,25 @@
+import csv
+
+from .models import Senateur
+
+
+def parse_senateurs(iterable):
+    lines = (
+        line
+        for line in iterable
+        if not line.startswith('%')
+    )
+    senateurs = [
+        Senateur(
+            qualite=row['Qualité'],
+            nom=row['Nom usuel'],
+            prenom=row['Prénom usuel'],
+            groupe=row['Groupe politique'],
+        )
+        for row in csv.DictReader(lines)
+    ]
+    by_name = {}
+    for s in senateurs:
+        by_name[f"{s.qualite} {s.nom}".upper()] = s
```

Comment:
Are we sure we never have homonyms?

Reply:
I started with last name only as a key, but some senators do have the same name.

I found that in the CSV file, the author is given by last name only when it's not ambiguous (`M. FRASSA`), and by first name + last name when it would be ambiguous (`M. DANIEL LAURENT`). So I added that format as a key too.

I trust them not to be ambiguous, but that might be too optimistic. If they use an ambiguous name (`M. LAURENT`) we might pick up the wrong one.

It would probably be safer to extract the unique senator ID from the URL in the `"Fiche Sénateur"` field (it looks like `"//www.senat.fr/senfic/frassa_christophe_andre08018u.html" and `08018U` can be used as a unique key to find the senator.

Reply:
Fixed in 81e83a3d9ba67b1bee687b260cdbc125de6448bc